contracts/AMM/ThalesAMM.sol

```diff
@@ -442,34 +416,36 @@ contract ThalesAMM is Initializable, ProxyOwned, ProxyPausable, ProxyReentrancyG
 
         address target = _getTarget(market, position);
 
+        address _liquidityPool = liquidityPool.getOrCreateMarketPool(market);
+
         //transfer options first to have max burn available
-        IERC20Upgradeable(target).safeTransferFrom(msg.sender, address(this), amount);
+        IERC20Upgradeable(target).safeTransferFrom(msg.sender, _liquidityPool, amount);
```

Comment:
This should be handled like it is done for the buy transactions. No need to add burnOnbehalf logic. Specifically:
1. Get all market positions from the LP contract into the AMM contract (see getOptions in buy methods)
2. Get all user market positions into the AMM contract
3. Now as AMM holds all the options, call burn directly from AMM
4. At the end send the sUSD and all remaining market options back to LP pool